### PR TITLE
Image generation improvements

### PR DESCRIPTION
**Pull request by Tymec**
### Background

[AUTOMATIC1111/stable-diffusion-webui](https://github.com/AUTOMATIC1111/stable-diffusion-webui) exposes an API for generating images using it. This feature can be useful for cutting down costs for generating images and allowing flexibility and more power for tweaking image generation. 
By using the API exposed by [AUTOMATIC1111/stable-diffusion-webui](https://github.com/AUTOMATIC1111/stable-diffusion-webui), we can generate images more efficiently and flexibly. This feature allows us to reduce the costs of image generation and to customize the image generation parameters for different models and settings.

### Changes

- Added support for using [AUTOMATIC1111/stable-diffusion-webui](https://github.com/AUTOMATIC1111/stable-diffusion-webui) API as an image provider
- Renamed "sd" image provider to "huggingface" to avoid confusion
- Added config options to change HuggingFace text-to-image model, image size, image count and SD WebUI settings
- Modified the `generate_image` function to accept arguments for prompt, negative prompt, num_images, image_size, model and extra params

### Documentation

- The code changes are documented using docstrings and comments

### Test Plan

- Created a unit test module for image generation to test each image provider

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [ ] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by Tymec** [comment]
I'm not sure whether adding the huggingface image model option breaks the atomicity of this pull request, so if there's any issues with that, please let me know.

**Review by nponeccop — DISMISSED** [review]

**Review by nponeccop — DISMISSED** [review]

**Comment by nponeccop** [comment]
@Tymec There are conflicts now

**Comment by nponeccop** [comment]
@Tymec CI is red

File: .gitignore
Comment (nponeccop) [inline .gitignore:1]: Is it related? If not revert

File: .pre-commit-config.yaml
Comment (nponeccop) [inline .pre-commit-config.yaml:1]: Clearly unrelated

File: autogpt/__main__.py [hunk 1]
```diff
@@ -18,13 +21,13 @@ def main() -> None:
     parse_arguments()
     logger.set_level(logging.DEBUG if cfg.debug_mode else logging.INFO)
     ai_name = ""
-    system_prompt = construct_prompt()
+    prompt = construct_prompt()
```
Comment (nponeccop) [inline autogpt/__main__.py]: Variable renaming isn't related to image generation improvements either

**Review by nponeccop — CHANGES_REQUESTED** [review]
Remove unrelated changes (not only those few I highlighted)

**Review by BillSchumacher — APPROVED** [review]

**Pull request merged**